SoftLayer/CLI/hardware/create_options.py

```diff
@@ -22,41 +26,171 @@ def cli(env):
-    tables.append(port_speed_table)
-
-    # Extras
-    extras_table = formatting.Table(['Extra Option', 'Value'], title="Extras")
-    extras_table.sortby = 'Value'
-    extras_table.align = 'l'
-    for extra in options['extras']:
-        extras_table.add_row([extra['name'], extra['key']])
-    tables.append(extras_table)
+    if location and prices:
+        raise exceptions.CLIAbort("Please select --prices or --location to get the prices, not both")
+
+    if prices:
+        _preset_prices_table(options['sizes'], tables)
+        _os_prices_table(options['operating_systems'], tables)
+        _port_speed_prices_table(options['port_speeds'], tables)
+        _extras_prices_table(options['extras'], tables)
+    elif location:
+        _preset_prices_table(options['sizes'], tables)
+        location_prices = hardware_manager.get_hardware_item_prices(location)
+        _location_item_prices(location_prices, tables)
+    else:
+        # Presets
+        preset_table = formatting.Table(['Size', 'Value'], title="Sizes")
+        preset_table.sortby = 'Value'
+        preset_table.align = 'l'
+        for size in options['sizes']:
+            preset_table.add_row([size['name'], size['key']])
+        tables.append(preset_table)
+
+        # Operating systems
+        os_table = formatting.Table(['OS', 'Key', 'Reference Code'], title="Operating Systems")
+        os_table.sortby = 'Key'
+        os_table.align = 'l'
+        for operating_system in options['operating_systems']:
+            os_table.add_row([operating_system['name'], operating_system['key'], operating_system['referenceCode']])
+        tables.append(os_table)
+
+        # Port speed
+        port_speed_table = formatting.Table(['Network', 'Speed', 'Key'], title="Network Options")
+        port_speed_table.sortby = 'Speed'
+        port_speed_table.align = 'l'
+        for speed in options['port_speeds']:
+            port_speed_table.add_row([speed['name'], speed['speed'], speed['key']])
+        tables.append(port_speed_table)
+
+        # Extras
+        extras_table = formatting.Table(['Extra Option', 'Value'], title="Extras")
+        extras_table.sortby = 'Value'
+        extras_table.align = 'l'
+        for extra in options['extras']:
+            extras_table.add_row([extra['name'], extra['key']])
+        tables.append(extras_table)
 
     env.fout(formatting.listing(tables, separator='\n'))
+
+
+def _preset_prices_table(sizes, tables):
+    """Shows Server Preset options prices.
+
+    :param [] sizes: List of Hardware Server sizes.
+    :param tables: Table formatting.
+    """
+    preset_prices_table = formatting.Table(['Size', 'Value', 'Hourly', 'Monthly'], title="Sizes Prices")
+    preset_prices_table.sortby = 'Value'
+    preset_prices_table.align = 'l'
+    for size in sizes:
+        preset_prices_table.add_row([size['name'], size['key'], size['hourlyRecurringFee'], size['recurringFee']])
```

Comment:
recurringFees should have as many decimals in their output as from the API output. Which I think is 4 decimal places.

Reply:
recurringFees seems to be all integer values, I did not see any recurringFees with decimal values by api output and printing this option by the code.
Please could you check for me this.

Reply:
Do you mean this?
![image](https://user-images.githubusercontent.com/39388146/91585084-91f27500-e921-11ea-9c75-be027a876138.png)
![image](https://user-images.githubusercontent.com/39388146/91585159-b1899d80-e921-11ea-88b4-b7f1675c2e94.png)

Reply:
Done.